patches/server/0964-Support-registry-mod-API-with-GameEvent.patch

```diff
@@ -0,0 +1,132 @@
+Date: Sat, 25 Feb 2023 21:25:08 -0800
+Subject: [PATCH] Support registry mod API with GameEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/registry/PaperRegistries.java b/src/main/java/io/papermc/paper/registry/PaperRegistries.java
+index be562b04c626c2fe8c792f445cbdd73a0bb5327f..198acd382c3059e040fc3cc8fd3f8748ab0efed4 100644
+--- a/src/main/java/io/papermc/paper/registry/PaperRegistries.java
++++ b/src/main/java/io/papermc/paper/registry/PaperRegistries.java
+@@ -2,6 +2,7 @@ package io.papermc.paper.registry;
+ 
+ import com.google.common.collect.ImmutableList;
+ import io.papermc.paper.registry.entry.RegistryEntry;
++import io.papermc.paper.world.PaperGameEvent;
+ import io.papermc.paper.world.structure.ConfiguredStructure;
+ import io.papermc.paper.world.structure.PaperConfiguredStructure;
+ import java.util.IdentityHashMap;
+@@ -20,6 +21,7 @@ import org.checkerframework.checker.nullness.qual.Nullable;
+ 
+ import static io.papermc.paper.registry.entry.RegistryEntry.immutable;
+ import static io.papermc.paper.registry.entry.RegistryEntry.immutableBuiltIn;
++import static io.papermc.paper.registry.entry.RegistryEntry.writableBuiltIn;
+ 
+ public final class PaperRegistries {
+ 
+@@ -32,6 +34,7 @@ public final class PaperRegistries {
+     static {
+         REGISTRY_ENTRIES = ImmutableList.<RegistryEntry<?, ?>>builder()
+             // built-ins
++            .add(writableBuiltIn(RegistryKey.GAME_EVENT, Registries.GAME_EVENT, PaperGameEvent::new, io.papermc.paper.world.PaperGameEvent.Builder::new))
+             .add(immutableBuiltIn(RegistryKey.STRUCTURE_TYPE, Registries.STRUCTURE_TYPE, CraftStructureType::new))
+ 
+             // data-drivens
+diff --git a/src/main/java/io/papermc/paper/world/PaperGameEvent.java b/src/main/java/io/papermc/paper/world/PaperGameEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..0995cd6008486c784072a21f1990c935a801f5bb
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/world/PaperGameEvent.java
+@@ -0,0 +1,59 @@
++package io.papermc.paper.world;
++
++import io.papermc.paper.registry.PaperRegistryBuilder;
++import org.bukkit.GameEvent;
++import org.bukkit.NamespacedKey;
++import org.checkerframework.checker.nullness.qual.NonNull;
++import org.checkerframework.checker.nullness.qual.Nullable;
++import org.checkerframework.framework.qual.DefaultQualifier;
++
++@DefaultQualifier(NonNull.class)
++public final class PaperGameEvent extends GameEvent {
++
++    private final net.minecraft.world.level.gameevent.GameEvent nms;
++
++    public PaperGameEvent(final NamespacedKey key, final net.minecraft.world.level.gameevent.GameEvent nms) {
++        super(key);
++        this.nms = nms;
++    }
++
++    @Override
++    public int getRange() {
++        return this.nms.getNotificationRadius();
++    }
++
++    public static final class Builder implements GameEvent.Builder, PaperRegistryBuilder<net.minecraft.world.level.gameevent.GameEvent, GameEvent> {
```

Comment:
Maybe include the vibration level inside the builder and the backed object ? The map seems to be static so i think it would be a good idea since there's no way otherwise. By default all the new game event will fallback with a vibration of zero.